packages/casadi/patches/0002-update-serializing_stream_hpp.patch

```diff
@@ -0,0 +1,40 @@
+From 86ac669c11f0f466ece32e6fcebacd62af0bdc32 Mon Sep 17 00:00:00 2001
+From: Agriya Khetarpal <74401230+agriyakhetarpal@users.noreply.github.com>
+Date: Mon, 15 Jul 2024 18:00:31 +0530
+Subject: [PATCH 2/4] Disable SIZE_MAX check for Emscripten, part I
+
+In WebAssembly (currently 32-bit), SIZE_MAX is equal to UINT_MAX, which meant
+these functions weren't being compiled, so there was no way to serialize or
+deserialize unsigned int values, leading to compile errors when these functions
+were called elsewhere. On many 32-bit systems, size_t and unsigned int are the
+same size, so SIZE_MAX == UINT_MAX is a common case.
+
+---
+ casadi/core/serializing_stream.hpp | 4 ++--
+ 1 file changed, 2 insertions(+), 2 deletions(-)
+
+diff --git a/casadi/core/serializing_stream.hpp b/casadi/core/serializing_stream.hpp
+index 47070d0e..e06081c1 100644
+--- a/casadi/core/serializing_stream.hpp
++++ b/casadi/core/serializing_stream.hpp
+@@ -99,7 +99,7 @@ namespace casadi {
+     void unpack(Slice& e);
+     void unpack(int& e);
+ 
+-#if SIZE_MAX != UINT_MAX
++#if SIZE_MAX != UINT_MAX || defined(__EMSCRIPTEN__)
```

Comment:
Again, please discuss with the upstream. It sounds to me that the correct approach might be to remove the `#if` entirely. Can you find out what the reason for including these in the first place was?

Reply:
I looked at the git blame to see if there's a reason, and this is the commit: https://github.com/casadi/casadi/commit/bff1521e726cbb3a2eaa009f572f5d1c0ba17c84 which claims to fix compatibility with 32-bit systems, which seems to make sense because on 64-bit systems, `size_t` is usually 64 bits, while `unsigned int` remains 32 bits. OTOH, on 32-bit systems, `size_t` and `unsigned int` are typically the same size, so without having this condition the compiler would ambiguously see two identical function signatures. Adding `__EMSCRIPTEN__` seemed to make the CasADi serialisation functionality work, so I left it like this.

Reply:
But `size_t` and `unsigned int` are the same size in Emscripten...

Reply:
I am confused about it, too – I shouldn't have needed to add `defined(__EMSCRIPTEN__)` here, but it fails without this as noted, plus WASM is 32-bit too. The associated blog I read doesn't talk much about this either: https://nachawati.me/blog/2023/08/08/compiling-casadi-webassembly/.